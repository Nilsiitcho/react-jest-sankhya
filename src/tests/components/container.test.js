import React from "react";
import {render, fireEvent} from "@testing-library/react";
import Counter from "../../components/counter";

test("Renderiza contador com o valor inicial de Zero", () =>{
   const {getByText} = render(<Counter/>)
   const counterElement = getByText(/0/i);
   expect(counterElement).toBeInTheDocument();
});

test("Renderiza contador corretamente", () => {
   const {asFragment} = render(<Counter />)
   expect(asFragment()).toMatchSnapshot();
})