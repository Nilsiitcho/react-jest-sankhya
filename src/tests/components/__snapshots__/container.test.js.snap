// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderiza contador corretamente 1`] = `
<DocumentFragment>
  <div
    class="counter-container"
  >
    <h1>
      Contador Sankhya
    </h1>
    <div
      class="container"
    >
      <button
        class="btn btn-decrease"
      >
        -
      </button>
      <span
        class="result "
      >
        0
      </span>
      <button
        class="btn btn-increase"
      >
        +
      </button>
    </div>
  </div>
</DocumentFragment>
`;
